### ISSUE TITLE
In iOS version 17.2, when moving the input focus in a WebView, an unknown area appears at the top of the screen.

### ISSUE THREAD

**Issue by wonbaejeon** [pichillilorenzo/flutter_inappwebview#1947]
- [x] I have read the [Getting Started](https://inappwebview.dev/docs/intro/) section
- [x] I have already searched for the same problem

## Environment

| Technology            | Version       |
|-----------------------| ------------- |
| Flutter version       |        (Channel stable, 3.16.4, on macOS 14.2 23C64 darwin-arm64,
locale ko-KR)       |
| Plugin version        |      flutter_inappwebview: ^5.7.2+3         |
| Android version       |         -      |
| iOS version           |          17.2     |
| macOS version         |         14.2      |
| Xcode version         |    15.1           |
| Google Chrome version |      -         |

Device information: 

## Description

**Expected behavior:** 

When moving the keyboard focus, there should be no unknown area appearing at the top of the screen

**Current behavior:** 

In iOS version 17.2, when moving the keyboard focus, an unknown area appears at the top of the screen

## Steps to reproduce

In iOS version 17.2, when moving the input focus in a WebView, an unknown area appears at the top of the screen.
Please refer to the video.

I am aware that setting 'resizeToAvoidBottomInset' of the scaffold to false prevents the UI from being resized by the keyboard, so the issue is not reproducible.
However, I want the bottom button to be positioned above the keyboard when the body size changed.

This issue is not reproducible on Android, and _**it occurs only on iOS version 17.2**._

I believe this may be an issue on the Flutter side due to the OS version update. Your thoughts on this matter are appreciated.

I observed different behaviors based on the iOS version when testing with the same code.

I released an app that implemented the sign-up screen in that way, but after the OS upgrade, the issue mentioned above is occurring

I am using Flutter InAppWebView, and the result is the same when using Flutter WebView
flutter_inappwebview: ^5.7.2+3

## Images   

![Monosnap screencast 2023-12-21 17-01-31](https://github.com/flutter/flutter/assets/45409353/f7da29fa-19f7-4645-9262-abc1b070d9a3)

**[Dart code]**
```dart
import 'package:flutter/material.dart';
import 'package:flutter_inappwebview/flutter_inappwebview.dart';

class TestWebView extends StatelessWidget {
  const TestWebView({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: SafeArea(
        child: InAppWebView(
          initialFile: "assets/test2.html",
        ),
      ),
    );
  }
}
```

**[WebCode]**
```javascript
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, height=device-height, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">
    <title>Mobile Keyboard Inputs</title>
    <script type="application/javascript" src="/assets/packages/flutter_inappwebview_web/assets/web/web_support.js" defer></script>
    <style>
    html,body{
        height:100%;
    }
    .container{
        position:relative;
        height:100%;
    }
    .btnBottom{
        position:fixed;
        width:100%;
        bottom:0;
        left:0;
    }
    </style>
</head>
<body>

<h2>Mobile Keyboard Inputs Example</h2>

<div class="container">
<form action="/submit" method="post">
        <label for="input0">Input Box 0:</label>
        <input type="text" id="input0" name="input0" placeholder="Type something..." />

        <br>

        <label for="input1">Input Box 1:</label>
        <input type="text" id="input1" name="input1" placeholder="Type something..." />

        <br>

        <label for="input2">Input Box 2:</label>
        <input type="number" id="input2" name="input2" pattern="\d*" placeholder="number" />

        <br>

        <label for="input3">Input Box 3:</label>
        <input type="number" id="input3" name="input3" pattern="\d*" placeholder="number" />

        <br>

        <button type="submit" class="btnBottom">버튼</button>
</form>
</div>

</body>
</html>

```

**Comment by hexh250786313** [comment]
Same on iOS 17.2.1.

**Comment by mohammedX6** [comment]
use latest version 6.0.0

**Comment by guzishiwo** [comment]
Same on iOS 17.3 beta

**Comment by AielloChan** [comment]
A smaller html file to reproduce this problem 

```html
<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" />
    <meta
      name="viewport"
      content="width=device-width,initial-scale=1,maximum-scale=1,user-scalable=0,viewport-fit=cover"
    />
  </head>
  <body>
    <div style="height: 100vh; overflow-y: auto">
      <div>Focus in different inputs to reproduce the problem</div>
      <input type="text" placeholder="focus me first" /> <br />
      <input type="text" placeholder="then focus me" /> <br />
      <button>hide keyboard</button>
      <div style="height: 1000px"></div>
    </div>
  </body>
</html>
```

**Comment by RMatushkin** [comment]
I think I can help you because I have faced strange focus behavior when a virtual keyboard appears, and I managed to find a solution that rid me of 99% of all problems, and it was always related to the SafeArea widget. Looking at its parameters, I found one called `maintainBottomViewPadding`, try setting its value to true, and I think all problems will go away.

**Comment by hexh250786313** [comment]
> Looking at its parameters, I found one called `maintainBottomViewPadding`, try setting its value to `true`, and I think all problems will go away.

@RMatushkin Is `maintainBottomViewPadding` the `SafeArea` parameter? I have tried to change its value but it can't help.

**Comment by hexh250786313** [comment]
Related flutter issue: https://github.com/flutter/flutter/issues/140501

**Comment by RMatushkin** [comment]
@hexh250786313 I don't see any issues on my side. Here is my environment:

> Flutter 3.16.5 • channel stable
Framework • revision 78666c8dc5 (2 weeks ago) • 2023-12-19 16:14:14 -0800
Engine • revision 3f3e560236
Tools • Dart 3.2.3 • DevTools 2.28.4

> flutter_inappwebview: 6.0.0

And this is how I use InAppWebView:
```
Scaffold(
  resizeToAvoidBottomInset: false,
  body: SafeArea(
    maintainBottomViewPadding: true,
    child: InAppWebView(...),
  ),
)
```

**Comment by wonbaejeon** [comment]
Dear @RMatushkin 
I am aware that setting 'resizeToAvoidBottomInset' of the scaffold to false prevents the UI from being resized by the keyboard, so the issue is not reproducible.
However, I want the bottom button to be positioned above the keyboard when the body size changed.

**Comment by RMatushkin** [comment]
> Dear @RMatushkin I am aware that setting 'resizeToAvoidBottomInset' of the scaffold to false prevents the UI from being resized by the keyboard, so the issue is not reproducible. However, I want the bottom button to be positioned above the keyboard when the body size changed.

I think I understand the problem. Yes, indeed, the bottom button is now completely blocked by my keyboard. My solution doesn't work, sorry.

**Comment by RMatushkin** [comment]
> @hexh250786313 I don't see any issues on my side. Here is my environment:
> 
> > Flutter 3.16.5 • channel stable • https://github.com/flutter/flutter.git
> > Framework • revision 78666c8dc5 (2 weeks ago) • 2023-12-19 16:14:14 -0800
> > Engine • revision 3f3e560236
> > Tools • Dart 3.2.3 • DevTools 2.28.4
> 
> > flutter_inappwebview: 6.0.0 • https://github.com/pichillilorenzo/flutter_inappwebview/releases/tag/v6.0.0
> 
> And this is how I use InAppWebView:
> 
> ```
> Scaffold(
>   resizeToAvoidBottomInset: false,
>   body: SafeArea(
>     maintainBottomViewPadding: true,
>     child: InAppWebView(...),
>   ),
> )
> ```
> 
>  753728e3-448a-42dd-87e4-d32b6e5f4c79.mp4

I tested the behavior in the Safari browser on a physical device. There behavior is exactly as I showed in this video here. Therefore, this behavior is standard. And in order to achieve the behavior that the author expects, it is necessary to write code and handle the keyboard, this is not a bug in the current WebView plugin.

**Comment by hexh250786313** [comment]
@RMatushkin Ty for your nice explanation, but I'm afraid I still don't know what happened. I am working on other projects recently and I don't have time to investigate this issue. My team decided to wait for the flutter official response.

I will try to figure it out when I have time. Awesome thanks for your help anyway.

**Comment by EdwardAW** [comment]
@RMatushkin It does appear that in the video you are using IOS version 17.0, not 17.2, which this bug doesn't appear to happen on.

**Comment by RMatushkin** [comment]
> @RMatushkin It does appear that in the video you are using IOS version 17.0, not 17.2, which this bug doesn't appear to happen on.

I mentioned above that the same behavior occurs in the Safari browser on a physical device with iOS version 17.2.1, when the keyboard covers the button. Could you please check this behavior on phusical device in Safari?

**Comment by crinity-dglee** [comment]
Is there any progress? 
I'm currently experiencing the same situation. As shown in the picture, the html element is moved to the bottom.

![스크린샷 2024-01-05 오전 10 53 17](https://github.com/pichillilorenzo/flutter_inappwebview/assets/109256013/6069ac39-c7a6-4654-a694-66722175520d)

**Comment by RMatushkin** [comment]
> Is there any progress? I'm currently experiencing the same situation. As shown in the picture, the html element is moved to the bottom.
> 
> ![스크린샷 2024-01-05 오전 10 53 17](https://private-user-images.githubusercontent.com/109256013/294363195-93f364b6-1714-4a4a-8b18-5c078cbe411a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ1NjM3NzYsIm5iZiI6MTcwNDU2MzQ3NiwicGF0aCI6Ii8xMDkyNTYwMTMvMjk0MzYzMTk1LTkzZjM2NGI2LTE3MTQtNGE0YS04YjE4LTVjMDc4Y2JlNDExYS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDEwNlQxNzUxMTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zMTg5YjY4NDcwZmJiYWM4M2Q4MjA0N2ZkZTAxNDA1ZDZkM2ViYjNhMmEzZTc3NmQ2ZmJiOTY3NDM1OTg0NmFlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Bqj-reSWEI4iLltz9JngFFOmj8nenS8rr5jmQIlmPrc)

Could you please check the behaviour on physical device in Safari? It seems like it's default behaviour in Safari, so I think we shouldn't expect another behaviour in WebView. As I wrote above, we have to write some part of the code to have this expected behaviour that described in this topic.

**Comment by dglee-me** [comment]
> > Is there any progress? I'm currently experiencing the same situation. As shown in the picture, the html element is moved to the bottom.
> > ![스크린샷 2024-01-05 오전 10 53 17](https://private-user-images.githubusercontent.com/109256013/294363195-93f364b6-1714-4a4a-8b18-5c078cbe411a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ1NjM3NzYsIm5iZiI6MTcwNDU2MzQ3NiwicGF0aCI6Ii8xMDkyNTYwMTMvMjk0MzYzMTk1LTkzZjM2NGI2LTE3MTQtNGE0YS04YjE4LTVjMDc4Y2JlNDExYS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDEwNlQxNzUxMTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zMTg5YjY4NDcwZmJiYWM4M2Q4MjA0N2ZkZTAxNDA1ZDZkM2ViYjNhMmEzZTc3NmQ2ZmJiOTY3NDM1OTg0NmFlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Bqj-reSWEI4iLltz9JngFFOmj8nenS8rr5jmQIlmPrc)
> 
> Could you please check the behaviour on physical device in Safari? It seems like it's default behaviour in Safari, so I think we shouldn't expect another behaviour in WebView. As I wrote above, we have to write some part of the code to have this expected behaviour that described in this topic.

It works normally in safari, but it only works like that in webview.

**Comment by dooully** [comment]
This issue occurred when IOS devices were updated to 17.3.

I tested both flutter_inappwebview and webview_flutter using the sample in the project, but the same problem occurred. 

The html code you loaded is the web code at the top.

This is a very important issue for me. 

I use text editor in webview, but I need to check that the keyboard is resized and show the toolbar. 

I pray that it will be resolved soon.

**Comment by holailusoria** [comment]
same issue

**Comment by lanistor** [comment]
This problem havn't been resolved till now.
Why iOS do this.

**Comment by wwiechorek** [comment]
I change my code to:

Scaffold(
        resizeToAvoidBottomInset: false,
        body: SafeArea(
          maintainBottomViewPadding: false, //try it to true if no work 
          child: WebViewWidget(controller: controller)
        ),
);


Original responde in: [https://github.com/flutter/flutter/issues/140501](https://github.com/flutter/flutter/issues/140501)

**Comment by neederkkh** [comment]
I have reproduced and tested the issue, and it seems to be an iOS-specific problem.  

cause : There's a part where scrollView.adjustedContentInset is applied to scrollView.contentInset. When the keyboard appears, the bottom value of contentInset becomes negative. In this scenario, if another input box is tapped, UIScrollView.contentOffset momentarily takes on the value that was applied to scrollView.contentInset.

solve : This is a temporary solution, but restoring scrollView.contentOffset in the aforementioned scenario resolves the issue for now. 

I've forked the repository and applied the fix there.

Fiile : pubspec.yaml  
```
dependency_overrides:
  flutter_inappwebview_ios:
    git:
      url: https://github.com/zpdL-Studio/flutter_inappwebview.git
      ref: 3ff3bb1681f41ffe49a9bdd8a1d7181014af7204
      path: flutter_inappwebview_ios
```

 It's a temporary fix, hopefully a permanent solution is found soon!

**Comment by holailusoria** [comment]
Same problem

**Comment by pichillilorenzo** [comment]
I finally found a workaround to fix this.
It will be available on the next beta version "6.2.0-beta.2".
In the meantime, I think you can test it using the master branch.

**Comment by pichillilorenzo** [comment]
Released new version `6.2.0-beta.2` with the fix.
